test/system/test_cvp_client_api.py

```diff
@@ -221,6 +221,119 @@ def test_api_user_operations(self):
         self.assertIn('total', result)
         self.assertEqual(result['total'], start_total)
 
+    def test_api_role_operations(self):
+        ''' Verify get_role, add_role and update_role
+        '''
+        # pylint: disable=too-many-statements
+        # pylint: disable=too-many-branches
+        # Test Get All Roles
+        result = self.api.get_roles()
+        self.assertIsNotNone(result)
+        self.assertIn('total', result)
+        start_total = result['total']
+
+        # Test Create Role
+        result = self.api.add_role('test_cvp_role', 'role description',
+                                    [{'name': 'image', 'mode': 'rw'}])
+        self.assertIsNotNone(result)
+        self.assertIn('data', result)
+        self.assertIn('key', result['data'])
+        self.assertIsNotNone(result['data']['key'])
+        self.assertIn('name', result['data'])
+        self.assertEqual(result['data']['name'], 'test_cvp_role')
+        self.assertIn('description', result['data'])
+        self.assertEqual(result['data']['description'], 'role description')
+        self.assertIn('createdBy', result['data'])
+        self.assertEqual(result['data']['createdBy'], 'cvpadmin')
```

Comment:
Likely don't want to hardcode the username check here since it is configurable in cvp_nodes.yaml.

Reply:
changed the test to just check that the createdBy is not none.